app/digests/api.py

```diff
@@ -28,15 +28,30 @@ class DigestViewSet(viewsets.ModelViewSet):
     content_type = settings.DIGEST_CONTENT_TYPE
     list_content_type = settings.DIGESTS_CONTENT_TYPE
 
+    def _can_modify(self) -> bool:
+        return self.request.META.get(settings.KONG_MODIFICATION_HEADER, False)
+
     def _create_response(self, data: Dict[str, Any]) -> Response:
         return Response(data, content_type=self.content_type)
 
+    def _is_authenticated(self) -> bool:
+        return self.request.META.get(settings.KONG_AUTH_HEADER, False)
+
     @staticmethod
     def _validate_against_schema(request: Request, data: Dict) -> None:
         schema = get_schema(get_schema_name(request.content_type))
         validate_json(data, schema=schema)
 
+    def get_queryset(self):
+        if self._is_authenticated():
+            return Digest.objects.all()
+        else:
+            return Digest.objects.filter(stage=PUBLISHED)
+
     def create(self, request: Request, *args, **kwargs) -> Response:
+        if not self._can_modify() or not self._is_authenticated():
```

Comment:
these two write-side methods could check only `_can_modify()`, for simpler isolation of permissions